common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -643,17 +637,58 @@ def get_problem_html(self, encapsulate=True):
         demand_hints = self.lcp.tree.xpath("//problem/demandhint/hint")
         demand_hint_possible = len(demand_hints) > 0
 
+        # Get the current problem status and generate the answer notification and message.
+        answer_notification_message = None
+        answer_notification_type = None
+        progress = self.get_progress()
+
+        if progress is not None:
+            key_list = self.correct_map.keys()
+            if len(key_list) == 1:
+                # Only one answer available or a missing progress and 1 or more answers.
+                answer_notification_type = self.correct_map.get(key_list[0]).get('correctness', None)
+            elif len(key_list) > 1:
+                # Look at the progress as we only want to show one notification for the problem. In this case there
+                # can be more that one correctness state in the correct_map.
+                if 0.0 < progress.percent() < 100.0:
+                    answer_notification_type = 'partially-correct'
+                elif progress.percent() == 100.0:
+                    answer_notification_type = 'correct'
+                else:
+                    answer_notification_type = 'incorrect'
+
+            ungettext = self.runtime.service(self, "i18n").ungettext
+            point_msg = ''
+            if progress is not None:
+                point_msg = ungettext(
+                    "({progress} point earned)",
+                    "({progress} points earned)",
+                    progress.frac()[1]
+                ).format(progress=str(progress))
+
+            # Build the notification message based on the notification type.
+            if answer_notification_type == 'incorrect':
+                answer_notification_message = 'Incorrect %s' % point_msg
```

Comment:
In order for the scraping to work for i18n, the "_" call has to happen where there is a literal string, not a variable. So even though _ is called with answer_notification_message, the message will only end up being translated if the exact text in the message happens to be in the translation dictionary (from someplace else where it is properly scraped). Not to mention the fact that part of the message is dynamic here...

The scraping has to happen here, with placeholders for the variable text.